test/unit/sdam/monitor.test.ts

```diff
@@ -306,6 +307,47 @@ describe('monitoring', function () {
         }, minHeartbeatFrequencyMS);
       });
     });
+
+    describe('roundTripTime', function () {
+      const table = [
+        {
+          serverMonitoringMode: 'stream',
+          topologyVersion: {
+            processId: new ObjectId(),
+            counter: new Long(0, 0)
+          }
+        },
+        { serverMonitoringMode: 'poll', topologyVersion: undefined }
+      ];
+      for (const { serverMonitoringMode, topologyVersion } of table) {
+        context(`when serverMonitoringMode = ${serverMonitoringMode}`, () => {
+          context('when more than one heartbeatSucceededEvent has been captured', () => {
+            const heartbeatDurationMS = 250;
+            it('correctly returns the mean of the heartbeat durations', async () => {
+              mockServer.setMessageHandler(request => {
+                setTimeout(
+                  () => request.reply(Object.assign({ helloOk: true }, mock.HELLO)),
+                  heartbeatDurationMS
+                );
+              });
+              const server = new MockServer(mockServer.address());
+              if (topologyVersion) server.description.topologyVersion = topologyVersion;
+              monitor = new Monitor(server as any, { serverMonitoringMode } as any);
+              monitor.connect();
+
+              for (let i = 0; i < 5; i++) {
+                await once(monitor, 'serverHeartbeatSucceeded');
+                monitor.requestCheck();
+              }
+              expect(monitor.roundTripTime).to.be.greaterThanOrEqual(heartbeatDurationMS);
```

Comment:
IIUC the test always uses the same delay, can we use a variation of delays so that if this stops being an average it would fail?

Reply:
So something like giving it a 100ms, 200ms, 300ms, 400ms, 500ms and checking the average should be around 750ms?